### PR TITLE
Playwright: Add even more info to the docs!

### PR DESCRIPTION
**Pull request by dpasque**
#### Changes proposed in this Pull Request

This adds some various ideas for documentation that I've accumulated over time to the Playwright E2E docs!

#### Testing instructions

Review docs - let me know if there's anything you don't think should be added!

[Link to parent doc on branch to view in browser.](https://github.com/Automattic/wp-calypso/tree/pw/more-markdown-updates/test/e2e)

Related to #

### CODE REVIEW

File: test/e2e/docs/patterns_tricks_gotchas.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	return 'foo';
+}
+
+// In the test spec...
+it( 'Active tab is "foo"', async function () {
+	expect( await pomClass.getActiveTabName() ).toBe( 'foo' );
+} );
+```
+
+**Be Warned!** This is not always as easy as it seems! Many of Playwright's functions that check state (like `isVisible()`) return _immediately_. This can be really tricky with async React apps like Calypso, which often require some measure of waiting for state to be propogated and components to be rendered.
```
Comment (worldomonation) [inline test/e2e/docs/patterns_tricks_gotchas.md]: I'm surprised the `**Be Warned**` part didn't fail linting, something about not using bold to do headings.

File: test/e2e/docs/style_guide.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <div style="width: 45%; float:left" align="left"><a href="./library_objects.md"><-- Library objects</a> </div>
 <div style="width: 5%; float:left" align="center"><a href="./../README.md">Top</a></div>
-<div style="width: 45%; float:right"align="right"><a href="./debugging.md">Debugging --></a> </div>
+<div style="width: 45%; float:right"align="right"><a href="./patterns_tricks_gotchas.md">Patterns, Tricks, and Gotchas --></a> </div>
```
Comment (worldomonation) [inline test/e2e/docs/style_guide.md:3]: Thanks for updating these links...they are not the most maintainable.

File: test/e2e/docs/tests_local.md [hunk 3]
```diff
@@ -69,6 +70,12 @@ Specify accounts to be pre-authenticated by saving authentication cookies and re
 export AUTHENTICATE_ACCOUNTS=simpleSitePersonalPlanUser,eCommerceUser,defaultUser
 ```
 
+### Use the mobile viewport
+
+```
+VIEWPORT_NAME=mobile yarn jest ...
+```
+
```
Comment (worldomonation) [inline test/e2e/docs/tests_local.md]: Can we generalize this a bit more, to something like:

```
### Specify viewports

By default, tests run against the `desktop` viewport size, approximately 1920x1080. The following viewports are currently supported:
- mobile
- desktop

To specify a viewport:
VIEWPORT_NAME=mobile yarn jest ...
```

**Review by worldomonation — APPROVED** [review]

File: test/e2e/docs/patterns_tricks_gotchas.md (hunk 1 above)
Comment (dpasque) [reply test/e2e/docs/patterns_tricks_gotchas.md]: Ah sure, I removed the bolding! 👍

**Pull request merged**